### PR TITLE
Don't break down chunks into multiple Amplitude API calls

### PR DESCRIPTION
**Pull request by willbarrett**
Let's just make each chunk small enough (via the config) that it won't trip any errors on Amplitude's side.

@robdiciuccio @change/data-science @change/platform

### CODE REVIEW

File: fluent-plugin-amplitude.gemspec [hunk 1]
```diff
@@ -4,7 +4,7 @@ $LOAD_PATH.unshift(lib) unless $LOAD_PATH.include?(lib)
 
 Gem::Specification.new do |spec|
   spec.name          = 'fluent-plugin-amplitude'
-  spec.version       = '0.2.3'
+  spec.version       = '0.2.4'
```
Comment (willbarrett) [inline fluent-plugin-amplitude.gemspec]: Whups - 0.2.4 has already been published. I'll update this on master.

File: lib/fluent/plugin/out_amplitude.rb [hunk 2]
```diff
@@ -165,40 +165,35 @@ def extract_event_properties!(amplitude_hash, record)
...
-      end
-      raise AmplitudeError, "Errors: #{errors_string}"
+    def log_error(error)
+      @statsd.track(
+        'fluentd.amplitude.records_errored',
+        records.length
+      )
+      error_string = "Response: #{error.code}, Time: #{error.time}, Body: #{error.body}, Duration: #{error.duration}"
+      log.error(error_string)
+      raise AmplitudeError, "Error: #{errors_string}"
```
Comment (robdiciuccio) [inline lib/fluent/plugin/out_amplitude.rb]: Why is the error raised in the logging method? Can we move this up?

File: lib/fluent/plugin/out_amplitude.rb [hunk 3]
```diff
@@ -165,40 +165,35 @@ def extract_event_properties!(amplitude_hash, record)
...
-    def log_errors(errors)
-      return if errors.empty?
-      errors_string = errors.map do |code, body, records, time|
-        "Response: #{code}, Time: #{time}, Body: #{body}, Record: #{records}"
-      end
-      raise AmplitudeError, "Errors: #{errors_string}"
+    def log_error(error)
+      @statsd.track(
+        'fluentd.amplitude.records_errored',
+        records.length
```
Comment (robdiciuccio) [inline lib/fluent/plugin/out_amplitude.rb]: `records` is not accessible in this scope

Comment (kamalama) [inline lib/fluent/plugin/out_amplitude.rb]: Can we pass this in by error code? that would be a nice plus - but not required.

**Pull request merged**